CMakeLists.txt

```diff
@@ -23,23 +23,23 @@ endif()
 
 # --------------------- Processor tests -------------------------
 
-message(STATUS "Building MLX for ${CMAKE_HOST_SYSTEM_PROCESSOR} processor on ${CMAKE_SYSTEM_NAME}")
+message(STATUS "Building MLX for ${CMAKE_SYSTEM_PROCESSOR} processor on ${CMAKE_SYSTEM_NAME}")
 
 set(MLX_BUILD_ARM OFF)
 
 if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-  if (${CMAKE_HOST_SYSTEM_PROCESSOR} MATCHES "x86_64" AND ${CMAKE_HOST_APPLE})
-    message(FATAL_ERROR
-      "Building for x86_64 on macOS is not supported."
-      " If you are on an Apple silicon system, check the build"
-      " documentation for possible fixes: "
-      "https://ml-explore.github.io/mlx/build/html/install.html#build-from-source")
-  elseif (${CMAKE_HOST_SYSTEM_PROCESSOR} MATCHES "x86_64")
-    message(WARNING
-      "Building for x86_64 on macOS is not supported."
-      " If you are on an Apple silicon system, "
-      " make sure you are building for arm64.")
-  elseif(${CMAKE_HOST_SYSTEM_PROCESSOR} MATCHES "arm64")
+  if(${CMAKE_SYSTEM_PROCESSOR} MATCHES "x86_64")
+    if(${CMAKE_HOST_SYSTEM_PROCESSOR} MATCHES "arm64" AND ${CMAKE_HOST_APPLE})
+      message(FATAL_ERROR
+        "Building for x86_64 on arm64 macOS is not supported."
+        " If you are on an Apple silicon system, check the build"
+        " documentation for possible fixes: "
+        "https://ml-explore.github.io/mlx/build/html/install.html#build-from-source")
+    else()
+      message(WARNING "Building for x86_64 arch is not officially supported.")
+    endif()
+    set(MLX_BUILD_METAL OFF)
```

Comment:
I'm cool enabling an x86 build if it's useful for you. That said, I think we have to make it explicitly opt-in. Too many people were accidentally building for x86 (e.g. through rosetta) before and the warning was not sufficient to alert them to the fact that their environment was bad.

What we could do is have a flag that you have to set with cmake asking for x86 `MLX_ENABLE_X86` or something..

Reply:
If there are people running cmake with rosetta, then the `CMAKE_HOST_SYSTEM_PROCESSOR` check won't work. ☹️ 

I have added a `MLX_ENABLE_X64_MAC` switch.